src/main/java/core/mate/academy/service/impl/MachineServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.mate.academy.service.impl;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import core.mate.academy.service.BulldozerProducer;
+import core.mate.academy.service.ExcavatorProducer;
+import core.mate.academy.service.MachineService;
+import core.mate.academy.service.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<Machine> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
```
Клас не обов'язково параметризовувати, адже в нас є інтерфейс, який уже вказує з яким типом ми будемо працювати